config/database.go

```diff
@@ -234,33 +236,47 @@ func (ds *DatabaseStore) persist(cfg *model.Config) error {
 	}
 	defer func() {
 		// Rollback after Commit just returns sql.ErrTxDone.
-		if err := tx.Rollback(); err != nil && err != sql.ErrTxDone {
+		if err = tx.Rollback(); err != nil && err != sql.ErrTxDone {
 			mlog.Error("Failed to rollback configuration transaction", mlog.Err(err))
 		}
 	}()
 
+	sum := sha256.Sum256(b)
 	params := map[string]interface{}{
 		"id":        id,
 		"value":     value,
 		"create_at": createAt,
 		"key":       "ConfigurationId",
+		"sha":       hex.EncodeToString(sum[0:]),
 	}
 
 	// Skip the persist altogether if we're effectively writing the same configuration.
-	var oldValue []byte
-	row := ds.db.QueryRow("SELECT Value FROM Configurations WHERE Active")
-	if err := row.Scan(&oldValue); err != nil && err != sql.ErrNoRows {
+	var oldValue string
+	row := ds.db.QueryRow("SELECT SHA FROM Configurations WHERE Active")
```

Comment:
Query plan for this please :)

Reply:
mysql:
| id | select_type | table          | partitions | type | possible_keys | key | key_len | ref | rows | filtered | Extra       |
|----|-------------|----------------|------------|------|---------------|-----|---------|-----|------|----------|-------------|
|  1 | SIMPLE      | Configurations |            | ALL  |               |     |         |     |    1 |   100.00 | Using where |

This is very interesting, even `EXPLAIN SELECT SHA FROM Configurations FORCE INDEX (Active) WHERE Active;` doesn't use the active index  `(╯°□°)╯︵ ┻━┻`

postgres:
| QUERY PLAN                                                                                         |
|----------------------------------------------------------------------------------------------------|
| Index Scan using configurations_active_key on configurations  (cost=0.15..8.17 rows=140 width=146) |
|   Index Cond: (active = true)                                                                      |
|   Filter: active                                                                                   |

Reply:
We go back to the old style then:

`select SHA from Configurations where Id = (select Id from Configurations Where Active);`